mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxFactory.kt

```diff
@@ -58,7 +56,7 @@ private fun <VM : BaseMvRxViewModel<S>, S : MvRxState> createViewModel(
     return requireNotNull(viewModel) {
         // If null, use Kotlin reflect for best error message. We will crash anyway, so performance
         // doesn't matter.
-        if (viewModelClass.kotlin.primaryConstructor?.parameters?.size?.let { it > 1 } == true) {
+        if (viewModelClass.constructors.firstOrNull()?.parameters?.size?.let { it > 1 } == true) {
```

Comment:
This seems to work fine and all tests pass.

Reply:
do we have a test where a State class has a secondary constructor?